### DISCUSSION TITLE
Adding data to TSCH EB causes issues when using Orcehstra scheduler

### DISCUSSION THREAD

**Discussion by alpealbert** [contiki-ng/contiki-ng#1685] [Q&A]
Hello, 
We are running into an issue that seems odd and we hope someone on this forum can help.
We want to add data to the enchanced beacons of our TSCH network. In order to do this, we have added a header type MLME_LONG_IE_TSCH_VENDOR_SPECIFIC_NESTED_IE which is used to identify the IE-packet and parse it correctly from the input buffer. 
Furthermore, we have extended the ieee802154_ies struct from frame802154e-ie.h to include additional data.

This seems to work well when runnning contiki-ng/examples/6tish/simple-node/node.c on the Z1 node platform without the Orchestra scheduler enabled.

However, when the Orchestra scheduler is enabled, testing a simple setup of a coordinator and a leaf node, the leaf node simply halts after a  few minutes.

Kind regards,

Oliver O. Nielsen
Master student at Department of applied Mathematics and Computer Science, Technical University of Denmark

**Comment by alexrayne** [comment]
> However, when the Orchestra scheduler is enabled, testing a simple setup of a coordinator and a leaf node, the leaf node simply halts after a few minutes

And without you additions, orchestra-mesh works well?

**Reply by alpealbert** [reply]
Hi, @alexrayne!
Yes, if we remove our contributions to the code, orchestra seems to function.
Without having precisely pinpointed the issue, it seems to have something to do with the expansion of the eee802154_ies struct. However, it is not clear to me _why_ this would affect the existing code, as we have only _added_ types to the struct and not modified the existing types.

**Reply by alexrayne** [reply]
it crushes silent? if turn on TSCH_LOG_PER_SLOT it prints any ineresting before crash?

**Reply by alexrayne** [reply]
i guess, orchestra stack just condition in wich other bug spoted

**Reply by alexrayne** [reply]
you debug on hardware or in cooja?

**Reply by alpealbert** [reply]
We are simulating in cooja, but the debugger only seems to be able to debug the simple-node.c file. Is it possible to debug the tsch*.c files with a debugger? If so - how?

**Reply by alexrayne** [reply]
in cooja sim, crashing of node will crash cooja without any info.
to help with it i provide [coffeecatch](https://github.com/contiki-ng/contiki-ng/pull/1225) exception hook. It intercepts SIGEV and if you lack - it dump stacktrace to log.

try it with [this cooja](https://github.com/alexrayne/cooja/tree/feature/jni-exception-handle), or better with [this one](https://github.com/alexrayne/cooja/tree/alexrayne). They have a bit handling node crashing - automaticaly stops simulation on it, and prints log with dumped stacktrace
